AmplifyPlugins/Core/AWSPluginsCore/Model/GraphQLDocument/Mutation/GraphQLDocument+CreateMutation.swift

```diff
@@ -8,48 +8,52 @@
 import Amplify
 import Foundation
 
-public extension GraphQLMutation {
-    convenience init(of anyModel: AnyModel, type mutationType: GraphQLMutationType) {
-        self.init(of: anyModel.instance, type: mutationType)
+public extension GraphQLCreateMutation {
+    convenience init(of anyModel: AnyModel,
```

Comment:
good question, i just tried to do this on the AppSync console:
```
mutation createPost($input:CreatePostInput!, $condition: ModelPostConditionInput) {
  createPost(input: $input, condition: $condition){
    content
    id
    title
  }
}

{
  "input": {
    "title": "title",
    "content": "content",
    "createdAt": "2011-11-11T11:11:11.111Z"
  },
  "condition": {
    "title": {
      "eq": "title"
    }
  }
}
```
 I can't get pass the 400 ConditionalCheckFailedException. What I think we should do here is:
- This GraphQLCreateMutation should not take in a predicate to avoid confusion on what the use case is ("conditionally update on an update/delete mutation").
- Model based GraphQL API's mutate takes in the predicate and mutation type. We can fail fast by doing client side validation.

Reply:
We need to go a bit deeper--why is the transformer generating create mutations with conditionals? Either it's a bug in the transformer, or there's a valid use case we're not catching. If the former, we need to get the CLI team to address. If the latter, we'll need to understand and support the use case.

Based on the [DDB docs](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.ConditionExpressions.html), and that fact that we're already explicitly setting a `attribute_not_exists` condition in the transformer-generated `onCreate*` resolvers, my belief is that conditional creates are not supported, and the transformer is incorrectly generating a mutation that allows a conditional input.

Reply:
thanks for looking into the DDB docs, will confirm that this is a bug with AppSync service team. If it is, then we should for sure do client validation or do not expose a way to do a create mutation with a condition (depending on how the APIs are structured). For example, mutation type can be an Enum with create, update(condition), delete(condition). or another way is as i was mentioning earlier, to do fail on the client with validation against a call to the Model-based API mutation with create mutation type and passing in the condition.

Reply:
re-read this comment and realized, this is amplify-cli transformation logic, which isn't AppSync service team's issue. Opening issue over in amplify-CLI repo https://github.com/aws-amplify/amplify-cli/issues/3146